data/pigui/modules/station-view/04-shipMarket.lua

```diff
@@ -180,29 +180,43 @@ function FormatAndCompareShips:beginTable()
 	return true
 end
 
-function FormatAndCompareShips:compare_and_draw_column(desc, str, a, b)
-	compare = a-b
+---@param desc       string        Description of the item being drawn
+---@param a	         number        The value for the new ship
+---@param b          number        The value for the old shio
+---@param fmt_a      function|nil  A function that takes a and formats it to a string, if required
+---@param fmt_a      function|nil  A function that takes b and formats it to a string, if nil then fmt_a is used
+function FormatAndCompareShips:compare_and_draw_column(desc, a, b, fmt_a, fmt_b)
+	local compare = a-b
 
 	if self.column == 0 then
 		ui.tableNextRow()
 	end
 
+	fmt_b = fmt_b and fmt_b or fmt_a
+
 	ui.tableSetColumnIndex(0 + self.column)
 	ui.textColored(styleColors.gray_300, desc)
 
 	ui.tableSetColumnIndex(1 + self.column)
 
+	local new_str = fmt_a and fmt_a( a ) or a
+
 	if compare < 0 then
-		ui.textAlignedColored(str, 1.0, ui.theme.colors.shipmarketCompareWorse)
-		ui.tableSetColumnIndex(2 + self.column)
-		ui.icon( ui.theme.icons.shipmarket_compare_worse, Vector2(ui.getTextLineHeight()), ui.theme.colors.shipmarketCompareWorse)		
-		
+		local old_str = fmt_b and fmt_b( b ) or b
+		ui.withTooltip( old_str, function ()
```

Comment:
Would prefer if this included a "Current Ship: " prefix to the text to clarify the context of the tooltip.

Reply:
As a note, if you've not already touched the translation system, you'd add a new entry to `data/lang/ui-core/en.json` in the format:
```json
"CURRENT_SHIP": {
    "description": "",
    "message": "Current Ship"
}
```

You'd use that new language string like so:
```lua
-- with string.interp shortcut operator '%'
ui.withTooltip( "{}: {}" % { l.CURRENT_SHIP, old_str }, function() --...
-- the boring (but slightly more performant) way
ui.withTooltip( l.CURRENT_SHIP .. ": " .. old_str, function() --...
```